### PR TITLE
Export memalign along with malloc and friends

### PR DESCRIPTION
**Pull request by dschuff**
Emscripten's mmap2 syscall started using memalign instead of malloc with https://github.com/kripken/emscripten/pull/4874, so we need to export that as well.

### CODE REVIEW

File: test/dot_s/export_malloc_free.s [hunk 1]
```diff
@@ -36,4 +36,22 @@ realloc:
 	i32.const	$push0=, 0
 	.endfunc
 .Lfunc_end22:
-	.size	realloc, .Lfunc_end22-free
```
Comment (jgravelle-google) [inline test/dot_s/export_malloc_free.s]: Is the `-free` part stripped and ignored, only used for human readability?

File: src/wasm-linker.cpp [hunk 2]
```diff
@@ -254,22 +254,19 @@ void Linker::layout() {
...
-  }
-  if (out.symbolInfo.implementedFunctions.count("realloc")) {
-    exportFunction("realloc", true);
+  for (auto function : {"malloc", "free", "realloc", "memalign"}) {
+    if (out.symbolInfo.implementedFunctions.count(function)) {
+      exportFunction(function, true);
+    }
   }
 
+
```
Comment (jgravelle-google) [inline src/wasm-linker.cpp]: Intentional extra newline?

File: test/dot_s/export_malloc_free.s (hunk 1 above)
Comment (dschuff) [inline test/dot_s/export_malloc_free.s]: It's generated for assemblers to calculate the size of objects, but It's ignored by s2wasm.

Comment (dschuff) [inline test/dot_s/export_malloc_free.s]: (the whole line is ignored, that is)

File: src/wasm-linker.cpp (hunk 2 above)
Comment (dschuff) [inline src/wasm-linker.cpp]: Unintentional, removed.

**Pull request merged**